### DISCUSSION TITLE
The stores could not be found

### DISCUSSION THREAD

**Discussion by SavArmadillo** [directus/directus#13205] [Archive]
Hello!

Using **npm** and **node 16**,

I am currently running into a problem while trying to create **custom layouts**.

I want to access the stores from `@directus/extensions-sdk` using this code:

```ts
// file: index.ts
import { useStores } from '@directus/extensions-sdk';

export default defineLayout<LayoutOptions, LayoutQuery>({

/* ... */

  setup(props, { emit }) {
    const { useFieldsStore, useCollectionsStore, usePresetsStore } = useStores();

    const fieldsStore = useFieldsStore();
    const collectionsStore = useCollectionsStore();
    const presetsStore = usePresetsStore();

  /* ... */
  };
});
```

So far, everything works. I can get all of the data stored in these store like the bookmarks from `presetsStore`.

But when I try to update data in these stores, for example:
```ts
//Still in the setup
presetsStore.update(presetTest.id, presetTest);
```

I get this error: 
![image](https://user-images.githubusercontent.com/91469954/167606143-93f493e3-0a7f-40be-81c9-c3d8d337da57.png)

So I am currently stuck in a sort of Read only mode for these stores witch is quite problematic.

I know there are two issues reffering to this error ([#9802](https://github.com/directus/directus/issues/9802) and [#8562](https://github.com/directus/directus/issues/8562)), but so far, they did'nt helped me. And with the fact these are month old, I was wondering if the solution changed too, since directus got updated.

**Comment by rijkvanzanten** [comment]
Could you share the full setup? I'm wondering if there's a particular timing problem in the usage of that `update` call 🤔

**Reply by SavArmadillo** [reply]
For now, that update is just right after I fetch the data:

```ts

setup(props, { emit }) {
  const { usePresetsStore } = useStores();
  const presetsStore = usePresetsStore();

  const presetTest = presetsStore.getPresetForCollection('X');

  //custom_metadata was created beforehand on directus
  presetTest.custom_metadata = {
    data1: 'foo',
    data2: 'bar'
  };

   console.log(presetsStore.update(presetTest.id, presetTest));

   //the rest of the setup function
}
```

I like to try out the different possibilities that creating custom extentions can offer before doing a real extention so I figured this was the easiest way to access and modify the data in the store.

But sadly, any attemps to change the data after reading it only resulted on the store not being found

**Reply by rijkvanzanten** [reply]
> But sadly, any attemps to change the data after reading it only resulted on the store not being found

That's the odd thing to me. If the store couldn't be found from the `useStores()` composable, than the `getPresetForCollection` call shouldn't work either 🤔

**Reply by SavArmadillo** [reply]
So I created a new custom layout extension to try and reproduce the problem and... it worked

I don't really understand what is going on here but i'll start from this new one while looking after that problem.
If I find / understand it and is somewhat useful to this project, i'll share my 'discovery' with an issue.

Thanks for your help!
